### ISSUE TITLE
Remove blindness on tombstone compaction with metrics

### ISSUE THREAD

**Issue by raphaelsc** [scylladb/scylladb#13742]
Today, we're completely blind about the tombstone purge process which happens in compaction.

Metrics can help us here, if we add metrics like the following:

total_tombstone_purge_attempt
total_tombstone_purge_success
total_tombstone_purge_failure_due_to_overlapping_with_memtable
total_tombstone_purge_failure_due_to_overlapping_with_uncompacting_sstable

**Comment by denesb** [comment]
I think we should also consider adding logging (or extend existing one) and/or extending the compaction history table.

**Comment by denesb** [comment]
@raphaelsc won't https://github.com/scylladb/scylladb/issues/7742 be a better fit? It is on-demand and it won't bloat the logs or metrics.

**Comment by raphaelsc** [comment]
> @raphaelsc won't #7742 be a better fit? It is on-demand and it won't bloat the logs or metrics.

you're right. monitoring is already slow enough and we can inspect sstable metadata and logs for knowing gc efficiency. I like your idea of extending history table, so we can see past efficiency on compactions that have completed already, it give you a sense of efficiency over time that I wanted when I opened this issue.

**Comment by denesb** [comment]
Turns out we already have an issue for embellishing compaction_history with additional data. I will close this as a duplicate and add the tombstone GC stats there.
